modules/hivemq-edge-module-modbus/src/main/java/com/hivemq/edge/adapters/modbus/ModbusProtocolAdapter.java

```diff
@@ -70,14 +75,18 @@ public ModbusProtocolAdapter(
         this.adapterConfig = adapterConfig;
         this.protocolAdapterState = input.getProtocolAdapterState();
         this.adapterFactories = input.adapterFactories();
+        this.modbusClient = new ModbusClient(adapterConfig, adapterFactories.dataPointFactory());
     }
 
     @Override
-    public void start(
-            @NotNull final ProtocolAdapterStartInput input, @NotNull final ProtocolAdapterStartOutput output) {
+    public void start(@NotNull final ProtocolAdapterStartInput input, @NotNull final ProtocolAdapterStartOutput output) {
         try {
-            initConnection();
-            output.startedSuccessfully();
+            modbusClient
+                    .connect()
+                    .thenRun(() -> {
```

Comment:
I think a exceptionally() is missing in case the connect() returns a failed future.

Reply:
or WhenComplete()